src/core/instance/render-helpers/render-slot.js

```diff
@@ -24,9 +24,9 @@ export function renderSlot (
       }
       props = extend(extend({}, bindObject), props)
     }
-    nodes = scopedSlotFn(props) || fallback
```

Comment:
We need to account for already render vnodes here (from component compiled with an older version of Vue)